gatewayconfig/processors/bluetooth_advertisement_processor.py

```diff
@@ -15,16 +16,42 @@
 class BluetoothAdvertisementProcessor:
     def __init__(self, eth0_mac_address, shared_state, variant_details):
         self.shared_state = shared_state
-        self.connection_advertisement = BluetoothConnectionAdvertisement(ADVERTISEMENT_INDEX, eth0_mac_address,
-                                                                         ADVERTISEMENT_TYPE, variant_details)
+        self.connection_advertisement = BluetoothConnectionAdvertisement(
+            ADVERTISEMENT_INDEX, eth0_mac_address,
+            ADVERTISEMENT_TYPE, variant_details)
+        self.stop_advertisement_timer = None
+
+    def start_advertisement(self):
+        logger.debug("Starting Bluetooth advertisement")
+        self.connection_advertisement.register()
+        self.shared_state.is_advertising_bluetooth = True
+        # for security, a start should always result in a scheduled stop
+        self.schedule_stop_advertisement()
+
+    def schedule_stop_advertisement(self, timer_seconds=ADVERTISEMENT_SECONDS):
+        if self.stop_advertisement_timer:
+            logger.debug("cancelling existing stop advertisement timer")
+            self.stop_advertisement_timer.cancel()
+
+        # trigger the time to stop advertisement
+        self.stop_advertisement_timer = threading.Timer(
+            timer_seconds, self.stop_advertisement)
+        self.stop_advertisement_timer.start()
+        logger.debug('scheduled stop advertisement in %s seconds', timer_seconds)
+
+    def stop_advertisement(self):
+        logger.debug("Stopping Bluetooth advertisement")
+        self.connection_advertisement.unregister()
+        self.shared_state.is_advertising_bluetooth = False
+        self.shared_state.should_scan_wifi = False
 
     def run(self):
         logger.debug("Running BluetoothAdvertisementProcessor")
 
         while True:
-            if(self.shared_state.should_advertise_bluetooth is True):
+            advertise_event = self.shared_state.should_advertise_bluetooth_condition_event.wait()
+            if advertise_event:
                 # Prepare to advertise by first scanning wifi and stopping any existing connections
-                self.shared_state.should_advertise_bluetooth = False
                 self.shared_state.should_scan_wifi = True
```

Comment:
Doesn't have anything to do with your changes, but this is confusing. I didn't realize/forgot this class is also responsible for setting the wifi scan flag. Can you create a new issue to address this weirdness. That ticket should include:

- Comments added to this class to clarify. Possibly changing the class name too.
- Triggering advertisement from some of the wifi related characteristics